app/test/qea-main-app-test.html

```diff
@@ -0,0 +1,73 @@
+<!doctype html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <script src="../../bower_components/webcomponentsjs/webcomponents-lite.js"></script>
+    <script src="../../bower_components/web-component-tester/browser.js"></script>
+    <script src="../../.transpiled/script/redux.min.js"></script>
+
+    <!-- import the element to test -->
+    <link rel="import" href="../../.transpiled/elements/elements.html">
+
+  </head>
+  <body>
+    <!-- use the document as a place to set up your fixtures -->
+    <test-fixture id="qea-main-app-fixture">
+      <template>
+        <qea-main-app ></qea-main-app>
+      </template>
+    </test-fixture>
+    <script>
+      suite('<qea-main-app>', function() {
+        var myEl;
+        setup(function() {
+          myEl = fixture('qea-main-app-fixture');
+          myEl._currentPage = 'landingPage';
+          stub('qea-main-app', {
+            resolveLocalPath: function(a) {
+              return '../../.transpiled/elements/'+a+'/'+a+'.html';
+            }
+          });
+        });
+
+        test('changePage event brings to the correct page', function() {
+          assert.equal(myEl.$$('iron-pages').selected, 'landingPage')
+          myEl.fire('changePage', {selectedPage: 'canvas'})
+          assert.equal(myEl.$$('iron-pages').selected, 'canvas')
+        });
+
+        test('paper-toast reacts to fireToast event', function () {
+          assert.equal(myEl.$$('qea-user-response').opened, false);
+          myEl.fire('fireToast', {text: 'test', type: 'error'});
+          assert.equal(myEl.$$('qea-user-response').opened, true);
+          assert.equal(myEl.$$('qea-user-response')._body, 'test');
+        })
+
+        // TODO the use of async or timeout function produce errors importing some bower components such as tinymce
```

Comment:
I have 2 pull request. in the latest one i remove some tests since I changed the structure of the main app component and also the imports are not done anymore in this component